### DISCUSSION TITLE
How to use other plugins in own plugin?

### DISCUSSION THREAD

**Discussion by meklis** [roadrunner-server/roadrunner#1542] [Q&A] [answered]
Hello! 
I want to create my plugin which would set the value in in-memory kv (php script will use these values).
I can't find examples of how to use other plugins in my.

Can you give an example how to access another plugin?

**Comment by rustatian** [comment]
Hey @meklis 👋🏻 
You can read docs here: https://roadrunner.dev/docs/customization-build/2023.x/en (and watch videos on this page). Feel free to ping me if you have any questions after that.

**Reply by meklis** [reply]
> Hey @meklis 👋🏻 You can read docs here: https://roadrunner.dev/docs/customization-build/2023.x/en (and watch videos on this page). Feel free to ping me if you have any questions after that.

Thanks for you reply. 
Yes, I watched the video, but it doesn't show how to include plugins as dependencies( 
And for plugin kv I can't found methods such as Get and Set (I don't understand how to set key-value)

**Reply by rustatian** [reply, marked as answer]
Here you are: https://youtu.be/w_uxFhdinvU?t=37😄
Basically, you can include the whole KV plugin, but this will lead to circular dependencies (not sure). 
In general, to include other plugins (or your own plugins), you need to declare the interface of that plugin in the plugin you want to use.

In your case this would be a new plugin, let's call it `super_plugin_1`.
In the `Init` method you need to use the interface of the `KV` plugin to include. For the second example you can look at the `Logger` plugin and the `Logger` interfaces from the tutorial.

But it would be better for you to create your own driver for the `KV` plugin. You can take a look at the [memory](https://github.com/roadrunner-server/memory) driver and use it as a base for your own plugin.
